tests/lib/rules/prefer-presence-queries.test.ts

```diff
@@ -837,6 +837,9 @@ ruleTester.run(RULE_NAME, rule, {
      // right after clicking submit button it disappears
      expect(submitButton).not.toBeInTheDocument()
     `,
+		`// checking absence on getBy* inside a within with queryBy* outside the within
+	 expect(within(screen.getByRole("button")).queryByText("Hello")).not.toBeInTheDocument()
```

Comment:
Could you add another test to check that chained queries from `within` are reported as expected tho?

```js
expect(within(screen.getByRole("button")).getByText("Hello")).not.toBeInTheDocument()
```

The example above should be reported, but I don't think it will be by your implementation.

Reply:
Good call. I think maybe I need to rethink the way I am handling this - it's not sufficient to say "ignore everything inside a within." Lemme rethink and re-PR.

Reply:
Actually I think this does correctly handle this use case. I've added the unit test as requested and will re-raise the PR - it correctly deals with this exact case.

Reply:
@Belco90 reopened at https://github.com/testing-library/eslint-plugin-testing-library/pull/717, with the additional test case